pkg/kubelet/lifecycle/handlers.go

```diff
@@ -123,8 +126,13 @@ func (hr *HandlerRunner) runHTTPHandler(pod *v1.Pod, container *v1.Container, ha
 			return "", err
 		}
 	}
-	url := fmt.Sprintf("http://%s/%s", net.JoinHostPort(host, strconv.Itoa(port)), handler.HTTPGet.Path)
-	resp, err := hr.httpGetter.Get(url)
+	u, err := url.Parse(handler.HTTPGet.Path)
+	if err != nil {
+		return "", err
+	}
+	u.Scheme = strings.ToLower(string(handler.HTTPGet.Scheme))
+	u.Host = net.JoinHostPort(host, strconv.Itoa(port))
+	resp, err := hr.httpGetter.Get(u.String())
```

Comment:
We should roundtrip `TLSClientConfig` here. WDYT?

Reply:
Yes, but I find httpGetter is an interface, and if we do not pass in as the previous, many test cases fail.